Sources/SentryCrash/Recording/Tools/SentryHook.c

```diff
@@ -56,6 +28,80 @@ sentrycrash__async_backtrace_decref(sentrycrash_async_backtrace_t *bt)
     }
 }
 
+/**
+ * This is a poor-mans concurrent hashtable.
+ * We have N slots, using modulo of the thread ID. Using atomic load / compare-exchange to make sure
+ * that the slot indeed belongs to the thread we want to work with.
+ */
+
+#define SENTRY_MAX_ASYNC_THREADS (128 - 1)
+
+typedef struct {
+    SentryCrashThread thread;
+    sentrycrash_async_backtrace_t *backtrace;
+} sentrycrash_async_caller_t;
+
+static sentrycrash_async_caller_t sentry_async_callers[SENTRY_MAX_ASYNC_THREADS] = { 0 };
+
+static size_t
+sentrycrash__thread_idx(SentryCrashThread thread)
+{
+    return (thread * 0x100000001b3 + 0xcbf29ce484222325) % SENTRY_MAX_ASYNC_THREADS;
```

Comment:
It is FNV, kind-of. In general the simple hash functions follow the `(a * x + b) mod c` pattern. It was just an attempt to hopefully better spread out the values. The `thread` here is a mach thread id, which should be a good number to begin with. I was experimenting with using the `pthread_t`, which itself is just a pointer which was giving worse results.

In general, since this is using a fixed number of slots, and does not account for collisions, you will get into problems with a high number of threads.

Reply:
Can we maybe add this info as a comment @Swatinem?